### PR TITLE
expose ulimits host_config-option

### LINKED ISSUE #34283: DockerOperator Expose `ulimits` create host config parameter
### Description

This issue should be resolved by simply adding `ulimits` parameter to `DockerOperator` that is directly passed to [`create_host_config`](https://docker-py.readthedocs.io/en/stable/api.html#docker.api.container.ContainerApiMixin.create_host_config).

### Use case/motivation

Currently applying custom `ulimits` is not possible with `DockerOperator`, making it necessary to do some hacky entrypoint workarounds instead.

By implementing this feature, one should be able to set Ulimits directly by giving a list of [`Ulimit`](https://docker-py.readthedocs.io/en/stable/api.html#docker.types.Ulimit)-instances to `DockerOperator`.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kahlstrm**
This is my first PR on this project, but mostly followed the format of the previous similar feature that exposed more `DockerOperator`-functionality. Tell me if something more is needed.
Closes: #34283 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: airflow/providers/docker/operators/docker.py
Comment (hussein-awala) [inline airflow/providers/docker/operators/docker.py:33]: Could you move it to `if TYPE_CHECKING` block?

**Review by hussein-awala — APPROVED** [review]
I think the static checks will fail (I hope so), LGTM after moving the imported class to `TYPE_CHECKING` condition.

File: airflow/providers/docker/operators/docker.py
Comment (kahlstrm) [reply airflow/providers/docker/operators/docker.py:33]: Will do, also will fix the failing tests I missed

Comment (kahlstrm) [reply airflow/providers/docker/operators/docker.py:33]: Hmm, ran the tests locally and static checks passed for me without `if TYPE_CHECKING`. Is it still necessary?

Comment (hussein-awala) [reply airflow/providers/docker/operators/docker.py:33]: If the static checks pass, no need for it, I will check it once the PR is merged

Comment (kahlstrm) [reply airflow/providers/docker/operators/docker.py:33]: Now spell checking is failing, so I'm adding `ulimit` to `spelling_wordlist.txt`.

**Pull request merged**
